monai/apps/mmars/config_resolver.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+import sys
+import warnings
+from importlib import import_module
+from typing import Any, Dict, List, Optional, Sequence
+
+from monai.apps.mmars.utils import search_configs_with_deps, update_configs_with_deps
+from monai.utils.module import instantiate
+
+__all__ = ["ComponentScanner", "ConfigComponent"]
+
+
+class ComponentScanner:
+    """
+    Scan all the available classes and functions in the MONAI package.
+    Map the all the names and the module names in a table.
+
+    Args:
+        excludes: if any string of the `excludes` exists in the full module name, don't import this module.
+
+    """
+
+    def __init__(self, excludes: Optional[Sequence[str]] = None):
+        self.excludes = [] if excludes is None else excludes
+        self._components_table = self._create_table()
```

Comment:
I usually prefer to put init logic in the constructor of a class to make the following logic `read-only`, in case we may want to support multi-thread parsing logic.
But anyway, we don't have such feature so far, updated your suggestion in the latest commit, it may help to optimize the performance especially for `lazy instantiation`.

Thanks.